PerimeterXModule/Internals/Helpers/RemoteServer.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Configuration;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace PerimeterX
+{
+	internal class RemoteServer
+	{
+		string _remoteUrl;
+		HttpContext _context;
+
+		/// <summary>
+		/// Initialize the communication with the Remote Server
+		/// </summary>
+		/// <param name="context">Context</param>
+		public RemoteServer(HttpContext context, string serverUrl, string uri)
+		{
+			_context = context;
+			_remoteUrl = serverUrl + uri;
+		}
+
+		/// <summary>
+		/// Return address to communicate to the remote server
+		/// </summary>
+		public string RemoteUrl
+		{
+			get
+			{
+				return _remoteUrl;
+			}
+		}
+
+		/// <summary>
+		/// Create a request the remote server
+		/// </summary>
+		/// <returns>Request to send to the server </returns>
+		public HttpWebRequest GetRequest()
+		{
+			CookieContainer cookieContainer = new CookieContainer();
+
+			// Create a request to the server
+			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(_remoteUrl);
+
+						// Set some options
+			var requestHeaders = _context.Request.Headers;
+			var proxyHeaders = new WebHeaderCollection();
+
+			foreach (string headerName in requestHeaders)
+			{
+				switch (headerName)
+				{
+					case "Connection":
+					case "Content-Length":
+					case "Date":
+					case "Expect":
+					case "Host":
+					case "If-Modified-Since":
+					case "Range":
+					case "Transfer-Encoding":
+					case "Proxy-Connection":
+						// Let IIS handle these
+					case "Accept":
+					case "Content-Type":
+					case "Referer":
+					case "User-Agent":
+						// Should be set after POST
+						break;
+
+					default:
+						proxyHeaders.Add(headerName, requestHeaders.Get(headerName));
+						break;
+				}
+			}
```

Comment:
You can simplify a code a bit to hold all these header names in a string array, then do something like 

```
var nonStandardHeaders = ["Connection", "Content-Length", ...];
foreach (string headerName in requestHeaders)
{
	if (!nonStandardHeaders.contains(headerName)) {
		proxyHeaders.Add(headerName, requestHeaders.Get(headerName));
	}
}
```

Reply:
👍